src/module_2/EDA_1.md

```diff
@@ -0,0 +1,3314 @@
+         99     eco_purchase_count = orders[orders['variant_id'].isin(eco_products['variant_id'])].shape[0]
+        101     total_eco_products = eco_products.shape[0]
+
+
+    File ~/.cache/pypoetry/virtualenvs/zrive-ds-O-AZ-S6U-py3.11/lib/python3.11/site-packages/pandas/core/frame.py:3767, in DataFrame.__getitem__(self, key)
+       3765     if is_iterator(key):
+       3766         key = list(key)
+    -> 3767     indexer = self.columns._get_indexer_strict(key, "columns")[1]
+       3769 # take() does not accept boolean indexers
+       3770 if getattr(indexer, "dtype", None) == bool:
+
+
+    File ~/.cache/pypoetry/virtualenvs/zrive-ds-O-AZ-S6U-py3.11/lib/python3.11/site-packages/pandas/core/indexes/base.py:5877, in Index._get_indexer_strict(self, key, axis_name)
+       5874 else:
+       5875     keyarr, indexer, new_indexer = self._reindex_non_unique(keyarr)
+    -> 5877 self._raise_if_missing(keyarr, indexer, axis_name)
+       5879 keyarr = self.take(indexer)
+       5880 if isinstance(key, Index):
+       5881     # GH 42790 - Preserve name from an Index
+
+
+    File ~/.cache/pypoetry/virtualenvs/zrive-ds-O-AZ-S6U-py3.11/lib/python3.11/site-packages/pandas/core/indexes/base.py:5938, in Index._raise_if_missing(self, key, indexer, axis_name)
+       5936     if use_interval_msg:
+       5937         key = list(key)
+    -> 5938     raise KeyError(f"None of [{key}] are in the [{axis_name}]")
+       5940 not_found = list(ensure_index(key)[missing_mask.nonzero()[0]].unique())
+       5941 raise KeyError(f"{not_found} not in index")
+
+
+    KeyError: "None of [Index([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,\n       ...\n       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan],\n      dtype='float64', length=1733)] are in the [columns]"
+
+
+
+```python
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+def check_abandoned_cart_prices(abandoned_cart, orders):
+    # Check data types
+    print("Abandoned Cart Data Types:")
+    print(abandoned_cart.dtypes)
+    print("\nOrders Data Types:")
+    print(orders.dtypes)
+    
+    # Check for NaN values
+    print("\nAbandoned Cart NaN Values:")
+    print(abandoned_cart.isna().sum())
+    print("\nOrders NaN Values:")
+    print(orders.isna().sum())
+
+    # Ensure that 'ordered_items' is numeric
+    orders['ordered_items'] = pd.to_numeric(orders['ordered_items'], errors='coerce')
+
+    # Calculate total price of abandoned carts
+    abandoned_cart_total_price = abandoned_cart.groupby('user_id').agg(total_price=('variant_id', 'count')).reset_index()
+    
+    # Calculate total price of converted carts
+    converted_cart_total_price = orders.groupby('user_id').agg(total_price=('ordered_items', 'sum')).reset_index()
+
+    # Merge results for analysis
+    merged = abandoned_cart_total_price.merge(converted_cart_total_price, on='user_id', suffixes=('_abandoned', '_converted'))
+
+    plt.figure(figsize=(12, 6))
```

Comment:
It is better if you decoupled the data prepation from the plotting. Or at least return the axes so whoever call this has the ability to build on top of the plot you provide